datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaConnectorTask.java

```diff
@@ -141,6 +141,7 @@ protected DatastreamProducerRecord translate(ConsumerRecord<?, ?> fromKafka, Ins
       metadata.put(BrooklinEnvelopeMetadataConstants.EVENT_TIMESTAMP, String.valueOf(readTime.toEpochMilli()));
       eventsSourceTimestamp = fromKafka.timestamp();
     }
+    metadata.put(BrooklinEnvelopeMetadataConstants.SOURCE_PARTITION, partitionStr);
```

Comment:
I wonder if _metadata.put("kafka-origin-partition", partitionStr);_ from line 129 is being used anywhere

Reply:
@celiakung at the moment I don't see it being used. Also, since metadata isn't propagated to the downstream Kafka, this data is lost. I did consider using "kafka-origin-partition", but since these are declared only for the Kafka connectors, I thought it'll make more sense to add a Metadata field which can be used by any connector that wants this  feature. Let me know if you think it may make sense to move out these metadata fields (all origin fields) to the common metadata constants. I wasn't sure since I wasn't sure why these are present.